src/diffpy/utils/tools.py

```diff
@@ -206,3 +209,56 @@ def get_package_info(package_names, metadata=None):
         pkg_info.update({package: importlib.metadata.version(package)})
     metadata["package_info"] = pkg_info
     return metadata
+
+
+def get_density_from_cloud(sample_composition, mp_token=""):
+    """Function to get material density from the MP database.
```

Comment:
add new function - not implemented

Reply:
Do we want to also go to the COD database (doesn't require a token).